akka-sse/src/main/scala/de/heikoseeberger/akkasse/LineParser.scala

```diff
@@ -34,15 +34,15 @@ private final class LineParser(maxLineSize: Int) extends GraphStage[FlowShape[By
 
   private val out = Outlet[String]("line-parser.out")
 
-  override val shape = FlowShape.of(in, out)
-
-  private var buffer = ByteString.empty
+  override val shape = FlowShape(in, out)
 
   override def createLogic(inheritedAttributes: Attributes) = new GraphStageLogic(shape) {
+    private var buffer = ByteString.empty
```

Comment:
Good point! I assume that `createLogic` is called for each materialization then and hence doesn't share any mutable state.